MoreLinq.Test/ReturnTest.cs

```diff
@@ -0,0 +1,145 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 Mitch Bodmer. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System;
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+    using System.Collections.Generic;
+
+    class ReturnTest
+    {
+        [Test]
+        public void TestResultingSequenceContainsSingle()
+        {
+            Assert.That(MoreEnumerable.Return(new object()).Count(), Is.EqualTo(1));
```

Comment:
The setup and casts look very verbose. Can we consolidate the test data? I see two patterns that can be encoded as nested classes:

```c#
        static class SomeSingleton
        {
            public static readonly object Item = new object();
            public static readonly IEnumerable<object> Sequence = MoreEnumerable.Return(Item);
            public static IList<object> List => (IList<object>)Sequence;
            public static ICollection<object> Collection => (ICollection<object>)Sequence;
        }

        static class NullSingleton
        {
            public static readonly IEnumerable<object> Sequence = MoreEnumerable.Return<object>(null);
            public static IList<object> List => (IList<object>)Sequence;
        }
```

then the test cases read simpler:

```c#
        [Test]
        public void TestCopyToSetsTheValueAtTheIndexToTheItemContained()
        {
            var array = new object[1];
            SomeSingleton.Collection.CopyTo(array, 0);
            Assert.That(array[0], Is.EqualTo(SomeSingleton.Item));
        }
```

Reply:
That's excellent!